File: Net/Api/src/main/java/org/elkoserver/foundation/net/LoadMonitor.java
package org.elkoserver.foundation.net;

/**
 * Interface implemented by classes that support tracking server load.
 */
public interface LoadMonitor {
    /**
     * Take note of some processing time spent.
     *
     * @param timeIncrement  The amount of processing time that was spent, in
     *    milliseconds.
     */
    void addTime(long timeIncrement);
}

File: Net/Api/src/main/kotlin/org/elkoserver/foundation/net/LoadMonitor.kt
package org.elkoserver.foundation.net

/**
 * Interface implemented by classes that support tracking server load.
 */
interface LoadMonitor {
    /**
     * Take note of some processing time spent.
     *
     * @param timeIncrement  The amount of processing time that was spent, in
     * milliseconds.
     */
    fun addTime(timeIncrement: Long)
}